src/string_map.c

```diff
@@ -354,8 +354,7 @@ rcutils_string_map_unset(rcutils_string_map_t * string_map, const char * key)
   size_t key_index;
   if (!__get_index_of_key_if_exists(string_map->impl, key, strlen(key), &key_index)) {
     char * msg = rcutils_format_string(allocator, "key '%s' not found", key);
-    RCUTILS_SET_ERROR_MSG(msg, allocator)
-    allocator.deallocate(msg, allocator.state);
+    rcutils_set_formatted_error(msg, allocator);
```

Comment:
This was ok before right?

Reply:
it doesn't check whether the memory allocated to `msg` is successful or not, just image that `msg` is **NULL** or a certain invalid memory address, and then to operate at that address.

Reply:
Makes sense.